packages/core-data/src/entity-types/post-status.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Internal dependencies
+ */
+import type { Context, OmitNevers } from './helpers';
+import type { BaseEntityRecords as _BaseEntityRecords } from './base-entity-records';
+
+declare module './base-entity-records' {
+	export namespace BaseEntityRecords {
+		/* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+		export interface PostStatusObject< C extends Context > {
+			/**
+			 * The title for the status.
+			 */
+			name: string;
```

Comment:
Aren't most (if not all) of these actually optional?

See:

https://github.com/WordPress/wordpress-develop/blob/80112f24ab0d0edce44097e4ed21ecdb933d8806/src/wp-includes/rest-api/endpoints/class-wp-rest-post-statuses-controller.php#L222-L286

Reply:
It's the same for user object but those fields are actually present because those are actually present in the item schema.

https://github.com/WordPress/wordpress-develop/blob/016bbeca9bfe96b3e024441059b76a3ad7d31df9/src/wp-includes/rest-api/endpoints/class-wp-rest-users-controller.php#L998-L1076